app/services/address_finder_service.rb

```diff
@@ -0,0 +1,27 @@
+class AddressFinderService
+  def initialize(postcode)
+    # Strip out non-alphanumeric characters
+    @postcode = postcode.gsub(/[^a-z0-9]/i, "")
+    @url = Rails.configuration.os_places_service_url + "/addresses.json?postcode=" + @postcode
+  end
+
+  def search_by_postcode
+    Rails.logger.debug "Sending request to OS Places service"
+
+    begin
+      response = RestClient::Request.execute(
+        method: :get,
+        url: @url
+      )
+
+      JSON.parse(response)
+    rescue RestClient::BadRequest
+      Rails.logger.debug "OS Places: resource not found"
```

Comment:
If OS places was down, would we fall into this rescue block? In which case it needs to send a notification toi Airbrake as well.

Reply:
I did find this example https://stackoverflow.com/questions/25552239/webmock-simulate-failing-api-no-internet-timeout but I think its one for another time hence marked PR as approved.